networkdb/delegate.go

```diff
@@ -63,10 +82,27 @@ func (nDB *NetworkDB) purgeSameNode(n *node) {
 }
 
 func (nDB *NetworkDB) handleNodeEvent(nEvent *NodeEvent) bool {
-	n := nDB.checkAndGetNode(nEvent)
+	// Update our local clock if the received messages has newer
+	// time.
+	nDB.networkClock.Witness(nEvent.LTime)
+
+	n := nDB.getNode(nEvent)
```

Comment:
I wonder if we can just search if the node is in the nodes list and ignore it if it is in the leftNodes and failedNodes